packages/web/src/components/premium-content-purchase-modal/components/PurchaseDetailsPage.tsx

```diff
@@ -85,6 +98,8 @@ export const PurchaseDetailsPage = ({
     )
   }, [isUnlocking, dispatch, track.track_id])
 
+  useNavigateOnSuccess(track, stage)
```

Comment:
super nit picky and no need to change, but I'm curious why you decided to split the hook out rather than using `useEffect` inline?

Reply:
I think I have an undocumented preference for components to not have _too much_ hook logic at the top and this seemed like a nice way to clearly indicate what the hook is doing and keep the component itself simple.